# Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview

| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |
| :----: | :---: | :----: | :--------------: |
| [GDCWholeGenomeSomaticSingleSample_v1.0.0](https://github.com/broadinstitute/warp/releases) | January, 2021 | [Elizabeth Kiernan](mailto:ekiernan@broadinstitute.org) | Please file GitHub issues in warp or contact [Kylee Degatano](mailto:kdegatano@broadinstitute.org) |

## Introduction to the GDC Whole Genome Somatic Single Sample Pipeline
Details of the references used and data harmonization process are found on the main page.

## Set-up

### Workflow Installation and Requirements

The workflow is written in the Workflow Description Language WDL and can be downloaded by cloning the [warp repository](https://github.com/broadinstitute/warp/tree/master) in GitHub. The workflow can be deployed using [Cromwell](https://github.com/broadinstitute/cromwell), a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. For the latest workflow version and release notes, please see the [changelog]().

### Software Version Requirements

* [GATK 4.1.8.0](https://github.com/broadinstitute/gatk/releases/tag/4.1.8.0)
* Picard 2.23.8
* Samtools 1.11
* Python 3.0
* Cromwell version support
    * Successfully tested on v52
    * Does not work on versions < v23 due to output syntax
* Papi version support
	* Successfully tested on Papi v2

### Input Requirements and Expectations

- Human whole genome sequencing data in unmapped BAM (uBAM) or CRAM format
- One or more read groups, one per uBAM/CRAM file, all belonging to a single sample (SM)
- Input uBAM/CRAM files must additionally comply with the following requirements:
    * Filenames all have the same suffix (we use ".unmapped.bam")
    * Files must pass validation by ValidateSamFile
    * Reads are provided in query-sorted order
    * All reads must have an RG tag
- Reference genome must be Hg38 with ALT contigs



## Workflow Tasks and Tools

The [workflow](https://github.com/broadinstitute/warp/blob/develop/beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl) imports a series of tasks either from the Workflow script or the Broad tasks library.

| Task Name  in WDL | Tool | Software and Version | Description | 
| --- | --- | --- | --- |
| [CramToUnmappedBams](https://github.com/broadinstitute/warp/blob/develop/tasks/broad/cram_to_unmapped_bams/CramToUnmappedBams.wdl) | --- | SamTools and Picard | If a CRAM file is used as input, the task converts to UBAM, generates an output map that is then used to split the uBAM by readgroup. The resulting BAM is sorted by query name using Picard |
| bam_readgroup_to_contents | --- | Samtools | Extracts all the readgroups from the BAM header and returns a WDL Array[Object] where each row is a readgroup. |
| biobambam_bamtofastq | --- | --- | Converts the uBAMs to FASTQ|
| emit_pe_records/emit_se_records | --- | --- | Associates the fasta file(s) generated by biobambam_bamtofastq with their respective readgroup; creates an Array of Structs to be used as input for alignment |
| bwa_pe/ bwa_se | --- | BWA mem and Samtools | Aligns FASTQ reads to the reference genome, generating an aligned BAM file |
| picard_markduplicates | [MarkDuplicates](https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-) | Picard | Runs MarkDuplicates in silent mode to locate and tag duplicate reads. Outputs a tagged BAM and metrics file. | 
| sort_and_index_markdup_bam | --- | Samtools | Sorts and indexes the BAM file; outputs the sorted BAM and index file | 
| gatk_baserecalibrator | [BaseRecalibrator](https://gatk.broadinstitute.org/hc/en-us/articles/360036898312) | GATK | Generates recalibration table for Base Quality Score Recalibration (BQSR) | 
| gatk_applybqsr | [ApplyBQSR](https://gatk.broadinstitute.org/hc/en-us/articles/360037055712) | GATK | Recalibrates the base qualities of the input reads based on the recalibration table produced by the BaseRecalibrator tool and outputs a recalibrated BAM file; uses the emit_original_quals parameter to write the original base qualities under the BAM OQ tag |
| collect_insert_size_metrics | [CollectInsertSizeMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360037055772) | Picard | Generates useful metrics about insert size distribution in the form of both a histogram and table (txt file) | 

## Workflow Outputs


## Important Notes

- Runtime parameters are optimized for Broad's Google Cloud Platform implementation.
- For help running workflows on the Google Cloud Platform or locally please
view the following tutorial [(How to) Execute Workflows from the gatk-workflows Git Organization](https://gatk.broadinstitute.org/hc/en-us/articles/360035530952).
- Please visit the [GATK Technical Documentation](https://gatk.broadinstitute.org/hc/en-us/categories/360002310591) site for further documentation on our workflows and tools.
- You can access relevant reference and resource bundles in the [GATK Resource Bundle](https://gatk.broadinstitute.org/hc/en-us/articles/360035890811).

## Contact Us


## Licensing

Copyright Broad Institute, 2020 | BSD-3

The workflow script is released under the **WDL open source code license (BSD-3)** (full license text at https://github.com/broadinstitute/warp/blob/master/LICENSE). However, please note that the programs it calls may be subject to different licenses. Users are responsible for checking that they are authorized to run all programs before running this script.

- [GATK](https://software.broadinstitute.org/gatk/download/licensing.php)
- [BWA](http://bio-bwa.sourceforge.net/bwa.shtml#13)
- [Picard](https://broadinstitute.github.io/picard/)
- [Samtools](http://www.htslib.org/terms/)


